common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -0,0 +1,101 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.verboseLog
+import java.io.IOException
+
+class DiagnosticsManager(
+    private val diagnosticsFileHelper: DiagnosticsFileHelper,
+    private val diagnosticsAnonymizer: DiagnosticsAnonymizer,
+    private val backend: Backend,
+    private val diagnosticsDispatcher: Dispatcher,
+    private val sharedPreferences: SharedPreferences
+) {
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val CONSECUTIVE_FAILURES_COUNT_KEY = "consecutive_failures_count"
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val MAX_NUMBER_RETRIES = 3
+
+        fun initializeSharedPreferences(context: Context): SharedPreferences =
+            context.getSharedPreferences(
+                "com_revenuecat_purchases_${context.packageName}_preferences_diagnostics",
+                Context.MODE_PRIVATE
+            )
```

Comment:
The plan is to retry if there is a network/500 error but we didn't want to retry forever. Since an immediate retry would not solve all cases, we use shared preferences to store the number of retries through app restarts. Lmk what you think of that logic!

Currently I'm creating a new shared preferences file for this. Once we migrate to use a separate shared preferences file for the whole SDK, we could unify and have a single one, but for now, I thought it best to not use the app's shared preferences.